### PR TITLE
feat(sca): add scan diff command

### PR DESCRIPTION
**Pull request by GG-HH**
Add scan diff command. This will allow users to perform a differential scan with custom reference.

### CODE REVIEW

File: tests/unit/cmd/sca/test_diff.py
Comment (amascia-gg) [inline tests/unit/cmd/sca/test_diff.py:1]: do we want to create a functional test as well?

File: tests/functional/sca/test_sca_scan_diff.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from pathlib import Path
+
+from tests.functional.utils import run_ggshield_sca_scan
+from tests.repository import Repository
+
+
+def test_scan_ci_diff(tmp_path: Path, monkeypatch, pipfile_lock_with_vuln) -> None:
```
Comment (amascia-gg) [inline tests/functional/sca/test_sca_scan_diff.py:7]:
```suggestion
def test_scan_diff(tmp_path: Path, monkeypatch, pipfile_lock_with_vuln) -> None:
```

Comment (amascia-gg) [reply tests/functional/sca/test_sca_scan_diff.py:7]: and it should work without `monkeypatch`

Comment (GG-HH) [reply tests/functional/sca/test_sca_scan_diff.py]: That is what happens when you copy paste a bit too fast... Sorry for that. I fixed it.

**Review by amascia-gg — APPROVED** [review]

**Pull request merged**